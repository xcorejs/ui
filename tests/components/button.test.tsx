import 'jest-styled-components';

import { Button } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Button component', () => {
  const component = renderer.create(
    <>
      <Button>Button</Button>
      <Button bg="navy">Button</Button>
      <Button v="outline">Button</Button>
      <Button s="lg">Button</Button>
      <Button s="lg" v="outline">Button</Button>
      <Button as="a" href="example.com">Link</Button>
      <Button disabled>Disabled</Button>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
