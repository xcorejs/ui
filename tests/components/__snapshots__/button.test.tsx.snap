// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component 1`] = `
Array [
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.1rem solid transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.7rem;
  padding-bottom: 0.7rem;
  font-size: 1.4rem;
  background: #0171b6;
  color: #fff;
}

.c0:hover {
  background: #0169a9;
}

.c0:active {
  background: #01619d;
}

.c0:focus {
  background: #01619d;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="background"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.4rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Button
  </button>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.1rem solid transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.7rem;
  padding-bottom: 0.7rem;
  font-size: 1.4rem;
  color: #fff;
  background: navy;
}

.c0:hover {
  background: #0169a9;
}

.c0:active {
  background: #01619d;
}

.c0:focus {
  background: #01619d;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="background"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.4rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Button
  </button>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.7rem;
  padding-bottom: 0.7rem;
  font-size: 1.4rem;
  border: 1px solid;
  border-color: #0171b6;
  color: #0171b6;
}

.c0:hover {
  background: #0171b6;
}

.c0:active {
  background: #0171b6;
}

.c0:focus {
  background: #0171b6;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="primary"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.4rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Button
  </button>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.1rem solid transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 3rem;
  padding-right: 3rem;
  padding-top: 1.2rem;
  padding-bottom: 1.2rem;
  font-size: 1.6rem;
  background: #0171b6;
  color: #fff;
}

.c0:hover {
  background: #0169a9;
}

.c0:active {
  background: #01619d;
}

.c0:focus {
  background: #01619d;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="background"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.6rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Button
  </button>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 3rem;
  padding-right: 3rem;
  padding-top: 1.2rem;
  padding-bottom: 1.2rem;
  font-size: 1.6rem;
  border: 1px solid;
  border-color: #0171b6;
  color: #0171b6;
}

.c0:hover {
  background: #0171b6;
}

.c0:active {
  background: #0171b6;
}

.c0:focus {
  background: #0171b6;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="primary"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.6rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Button
  </button>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.1rem solid transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.7rem;
  padding-bottom: 0.7rem;
  font-size: 1.4rem;
  background: #0171b6;
  color: #fff;
}

.c0:hover {
  background: #0169a9;
}

.c0:active {
  background: #01619d;
}

.c0:focus {
  background: #01619d;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<a
    className="c0"
    color="background"
    cursor="pointer"
    display="inline-flex"
    fontFamily="text"
    fontSize="1.4rem"
    fontWeight={500}
    href="example.com"
    role="group"
    textDecoration="none"
  >
    Link
  </a>,
  .c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 2.4rem;
  border-radius: 0.3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.1rem solid transparent;
  -webkit-transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  transition: background 300ms,color 300ms,border 300ms,box-shadow 300ms;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.7rem;
  padding-bottom: 0.7rem;
  font-size: 1.4rem;
  background: #0171b6;
  color: #fff;
}

.c0:hover {
  background: #0169a9;
}

.c0:active {
  background: #01619d;
}

.c0:focus {
  background: #01619d;
  outline: 2px solid rgba(15,31,40,0.2);
  outline-offset: -2px;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    className="c0"
    color="background"
    cursor="pointer"
    disabled={true}
    display="inline-flex"
    fontFamily="text"
    fontSize="1.4rem"
    fontWeight={500}
    role="group"
    textDecoration="none"
  >
    Disabled
  </button>,
]
`;
